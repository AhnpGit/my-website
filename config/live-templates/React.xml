<templateSet group="React">
  <template name="rx_module" value="const _ACTION_PREFIX = '$projectName$/$moduleName$'&#10;&#10;const $ACTION_NAME$ = `${_ACTION_PREFIX}/$ACTION_NAME$`&#10;&#10;// TODO: Connect the reducer.&#10;export default function reducer(state = {$defaultState$}, action) {&#10;    switch (action.type) {&#10;        case $ACTION_NAME$:&#10;            return {$END$}&#10;&#10;        default:&#10;            return state&#10;    }&#10;}&#10;&#10;export const $actionCreator$ = () =&gt; ({&#10;    type: $ACTION_NAME$,&#10;    payload: {$payload$},&#10;    meta: {},&#10;    error: undefined,&#10;})&#10;" description="Creates a DUCKS-module" toReformat="true" toShortenFQNames="true">
    <variable name="projectName" expression="" defaultValue="&quot;&lt;project name&gt;&quot;" alwaysStopAt="true" />
    <variable name="moduleName" expression="fileNameWithoutExtension()" defaultValue="" alwaysStopAt="true" />
    <variable name="ACTION_NAME" expression="" defaultValue="&quot;MY_ACTION&quot;" alwaysStopAt="true" />
    <variable name="defaultState" expression="" defaultValue="" alwaysStopAt="true" />
    <variable name="actionCreator" expression="" defaultValue="&quot;myActionCreator&quot;" alwaysStopAt="true" />
    <variable name="payload" expression="" defaultValue="" alwaysStopAt="true" />
    <context>
      <option name="JS_STATEMENT" value="true" />
    </context>
  </template>
  <template name="rx_component" value="import React from 'react'&#10;&#10;const $ComponentName$ = ({$props$}) =&gt; (&#10;    &lt;$END$/&gt;&#10;)&#10;&#10;export default $ComponentName$" description="Creates a functional component" toReformat="true" toShortenFQNames="true">
    <variable name="ComponentName" expression="capitalize(camelCase(fileNameWithoutExtension()))" defaultValue="" alwaysStopAt="true" />
    <variable name="props" expression="" defaultValue="" alwaysStopAt="true" />
    <context>
      <option name="JS_STATEMENT" value="true" />
    </context>
  </template>
  <template name="rx_component_class" value="import React from 'react'&#10;&#10;class $ComponentName$ extends React.Component {&#10;    render() {&#10;        return (&#10;            &lt;$END$/&gt;&#10;        )&#10;    }&#10;}&#10;&#10;export default $ComponentName$" description="Creates a class component" toReformat="true" toShortenFQNames="true">
    <variable name="ComponentName" expression="capitalize(camelCase(fileNameWithoutExtension()))" defaultValue="" alwaysStopAt="true" />
    <context>
      <option name="JS_STATEMENT" value="true" />
    </context>
  </template>
  <template name="rx_proptypes" value="import PropTypes from 'prop-types'&#10;$ComponentName$.propTypes = {&#10;    $prop$: PropTypes.$END$,&#10;}" description="Adds proptypes to a component" toReformat="true" toShortenFQNames="true">
    <variable name="ComponentName" expression="capitalize(camelCase(fileNameWithoutExtension()))" defaultValue="" alwaysStopAt="true" />
    <variable name="prop" expression="" defaultValue="&quot;prop&quot;" alwaysStopAt="true" />
    <context>
      <option name="JS_STATEMENT" value="true" />
    </context>
  </template>
  <template name="rx_action_async" value="const $ACTION_NAME$ = `${_ACTION_PREFIX}/$ACTION_NAME$`&#10;const $ACTION_NAME$_PENDING = `${$ACTION_NAME$}_PENDING`&#10;const $ACTION_NAME$_FULFILLED = `${$ACTION_NAME$}_FULFILLED`&#10;const $ACTION_NAME$_REJECTED = `${$ACTION_NAME$}_REJECTED`" description="Creates async action types" toReformat="true" toShortenFQNames="true">
    <variable name="ACTION_NAME" expression="" defaultValue="&quot;MY_ACTION&quot;" alwaysStopAt="true" />
    <context>
      <option name="JS_STATEMENT" value="true" />
    </context>
  </template>
  <template name="rx_action" value="const $ACTION_NAME$ = `${_ACTION_PREFIX}/$ACTION_NAME$`" description="Creates an action type" toReformat="true" toShortenFQNames="true">
    <variable name="ACTION_NAME" expression="" defaultValue="&quot;MY_ACTION&quot;" alwaysStopAt="true" />
    <context>
      <option name="JS_STATEMENT" value="true" />
    </context>
  </template>
  <template name="rx_creator" value="export const $actionCreator$ = () =&gt; ({&#10;    type: $ACTION_NAME$,&#10;    payload: {$END$},&#10;    meta: {},&#10;    error: undefined,&#10;})" description="Creates an action creator" toReformat="true" toShortenFQNames="true">
    <variable name="actionCreator" expression="" defaultValue="&quot;myActionCreator&quot;" alwaysStopAt="true" />
    <variable name="ACTION_NAME" expression="" defaultValue="&quot;MY_ACTION&quot;" alwaysStopAt="true" />
    <context>
      <option name="JS_STATEMENT" value="true" />
    </context>
  </template>
  <template name="rx_creator_async" value="export const $actionCreator$ = () =&gt; ({&#10;    type: $ACTION_NAME$,&#10;    payload: getPayload(),&#10;    meta: {},&#10;    error: undefined,&#10;    &#10;    async function getPayload() {&#10;        return $END$&#10;    }&#10;})" description="Creates an async action creator" toReformat="true" toShortenFQNames="true">
    <variable name="actionCreator" expression="" defaultValue="&quot;myActionCreator&quot;" alwaysStopAt="true" />
    <variable name="ACTION_NAME" expression="" defaultValue="&quot;MY_ACTION&quot;" alwaysStopAt="true" />
    <context>
      <option name="JS_STATEMENT" value="true" />
    </context>
  </template>
  <template name="rx_container" value="import { connect } from 'react-redux'&#10;import $ComponentName$ from '../components/$ComponentName$'&#10;&#10;const mapStateToProps = (state) =&gt; ({&#10;    $END$&#10;})&#10;&#10;const mapDispatchToProps = (dispatch) =&gt; ({&#10;})&#10;&#10;export default connect(&#10;    mapStateToProps,&#10;    mapDispatchToProps,&#10;)($ComponentName$)" description="Creates a container class" toReformat="true" toShortenFQNames="true">
    <variable name="ComponentName" expression="capitalize(camelCase(fileNameWithoutExtension()))" defaultValue="" alwaysStopAt="true" />
    <context>
      <option name="JS_STATEMENT" value="true" />
    </context>
  </template>
  <template name="rx_styled" value="import styled from 'styled-components'&#10;&#10;const $ComponentName$ = styled.$ElementType$`&#10;  $END$&#10;`&#10;&#10;export default $ComponentName$" description="Creates a styled component" toReformat="true" toShortenFQNames="true">
    <variable name="ComponentName" expression="capitalize(camelCase(fileNameWithoutExtension()))" defaultValue="" alwaysStopAt="true" />
    <variable name="ElementType" expression="" defaultValue="&quot;div&quot;" alwaysStopAt="true" />
    <context>
      <option name="JS_STATEMENT" value="true" />
    </context>
  </template>
</templateSet>
