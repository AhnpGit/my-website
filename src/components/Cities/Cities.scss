.wrapper__cities {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;

  @include media-query($medium-up) {
    padding: 0 $base-unit;
  }

  .cities__intro {
    @extend .wrapper__cities;
    padding: 0;
    margin-bottom: $base-unit;

    h2 {
      text-align: center;
      word-spacing: 2px;
      letter-spacing: 1px;
      text-transform: uppercase;

      &:after {
        display: block;
        height: 2px;
        background-color: $tahitiGold;
        content: " ";
        width: 100px;
        margin: 0 auto;
        margin-top: $base-unit;
      }
    }
  }

  .cities__body {
    lost-center: 100% 0 flex;
    .cities__features {
      lost-row: 1/4 $base-unit;
    }

    @include media-query($medium-up) {
      .cities__features {
        lost-column: 1/4 0 $base-unit;
      }
    }

    h3 {
      text-transform: uppercase;
    }
  }
}

.city__feature {
  display: flex;
  align-items: center;
  @include media-query($small-only) {
    font-size: 85%;
  }

  svg {
    fill: $tahitiGold;
    margin-right: 5px;
  }

  a {
    color: $trout;
  }
}
