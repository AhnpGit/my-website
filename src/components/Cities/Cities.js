import React from 'react'
import Icon from '../../elements/Icon/Icon'

const Cities = () => (
  <section className="cities">
    <div className="cities__intro">
      <h2>We're currently in these cities</h2>
    </div>
    <div className="cities__specs">
      <div className="cities__features">
        <img src="https://lh4.googleusercontent.com/PdKlSTipBojRngRyozz25bNcjSXDgpoPEjX0_6GRcZpz58aJMjklJg"
             alt="Lisbon" />
        <h3>Lisbon</h3>
        <div className="city__feature">
          <Icon name="profile" />
          1600+ happy eaters
        </div>
        <div className="city__feature">
          <Icon name="star" />
          60+ top chefs
        </div>
        <div className="city__feature">
          <Icon name="twitter" />
          <a href="">@omnifood_lx</a>
        </div>
      </div>
      <div className="cities__features">
        <img src="https://lh3.googleusercontent.com/lDQGvYX2seloj1UDjLnE1ooiOP-5DjUOIMvRtb3aOShF-nG_OpZWcw"
             alt="San Francisco" />
        <h3>San Francisco</h3>
        <div className="city__feature">
          <Icon name="profile" />
          3700+ happy eaters
        </div>
        <div className="city__feature">
          <Icon name="star" />
          160+ top chefs
        </div>
        <div className="city__feature">
          <Icon name="twitter" />
          <a href="">@omnifood_sf</a>
        </div>
      </div>
      <div className="cities__features">
        <img src="https://lh5.googleusercontent.com/HWe5CSBiGeieGvPoUsxUrcJO2hKeZILpn9CcM0WkcFps7C5Rp7LrTA"
             alt="Berlin" />
        <h3>Berlin</h3>
        <div className="city__feature">
          <Icon name="profile" />
          2300+ happy eaters
        </div>
        <div className="city__feature">
          <Icon name="star" />
          110+ top chefs
        </div>
        <div className="city__feature">
          <Icon name="twitter" />
          <a href="">@omnifood_berlin</a>
        </div>
      </div>
      <div className="cities__features">
        <img src="https://lh4.googleusercontent.com/IzAVIrEBuvthQbp2kFZMCz9c95jr593I650s81GuaWf2MFiGy3utjQ"
             alt="London" />
        <h3>London</h3>
        <div className="city__feature">
          <Icon name="profile" />
          1200+ happy eaters
        </div>
        <div className="city__feature">
          <Icon name="star" />
          50+ top chefs
        </div>
        <div className="city__feature">
          <Icon name="twitter" />
          <a href="">@omnifood_london</a>
        </div>
      </div>
    </div>
  </section>
)

export default Cities
