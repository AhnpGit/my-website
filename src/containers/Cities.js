// @flow

import { connect } from 'react-redux'
import Cities from '../components/Cities/Cities'
import type { CitiesType } from '../utils/types'

type CitiesState = {
  cities: CitiesType
}

export const mapStateToProps = (state : CitiesState) => ({
  cityItems: state.cities.cityItems
})

export default connect(
  mapStateToProps
)(Cities)
