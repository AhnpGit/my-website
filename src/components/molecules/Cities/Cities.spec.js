import React from 'react'
import { shallow } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'
import Cities from './Cities'

describe('Cities component', () => {
  it('renders correctly without any issues', () => {
    const component = shallow(
      <Cities />
    )

    const result = shallowToJson(component)
    expect(result).toMatchSnapshot()
  })
})
