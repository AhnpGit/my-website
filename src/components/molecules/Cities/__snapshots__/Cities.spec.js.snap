// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cities component renders correctly without any issues 1`] = `
<section
  className="cities"
>
  <div
    className="cities__intro"
  >
    <h2>
      We're currently in these cities
    </h2>
  </div>
  <div
    className="cities__specs"
  >
    <div
      className="cities__features"
    >
      <img
        alt="Lisbon"
        src="https://lh4.googleusercontent.com/PdKlSTipBojRngRyozz25bNcjSXDgpoPEjX0_6GRcZpz58aJMjklJg"
      />
      <h3>
        Lisbon
      </h3>
      <div
        className="city__feature"
      >
        <Icon
          name="profile"
        />
        1600+ happy eaters
      </div>
      <div
        className="city__feature"
      >
        <Icon
          name="star"
        />
        60+ top chefs
      </div>
      <div
        className="city__feature"
      >
        <Icon
          name="twitter"
        />
        <a
          href=""
        >
          @omnifood_lx
        </a>
      </div>
    </div>
    <div
      className="cities__features"
    >
      <img
        alt="San Francisco"
        src="https://lh3.googleusercontent.com/lDQGvYX2seloj1UDjLnE1ooiOP-5DjUOIMvRtb3aOShF-nG_OpZWcw"
      />
      <h3>
        San Francisco
      </h3>
      <div
        className="city__feature"
      >
        <Icon
          name="profile"
        />
        3700+ happy eaters
      </div>
      <div
        className="city__feature"
      >
        <Icon
          name="star"
        />
        160+ top chefs
      </div>
      <div
        className="city__feature"
      >
        <Icon
          name="twitter"
        />
        <a
          href=""
        >
          @omnifood_sf
        </a>
      </div>
    </div>
    <div
      className="cities__features"
    >
      <img
        alt="Berlin"
        src="https://lh5.googleusercontent.com/HWe5CSBiGeieGvPoUsxUrcJO2hKeZILpn9CcM0WkcFps7C5Rp7LrTA"
      />
      <h3>
        Berlin
      </h3>
      <div
        className="city__feature"
      >
        <Icon
          name="profile"
        />
        2300+ happy eaters
      </div>
      <div
        className="city__feature"
      >
        <Icon
          name="star"
        />
        110+ top chefs
      </div>
      <div
        className="city__feature"
      >
        <Icon
          name="twitter"
        />
        <a
          href=""
        >
          @omnifood_berlin
        </a>
      </div>
    </div>
    <div
      className="cities__features"
    >
      <img
        alt="London"
        src="https://lh4.googleusercontent.com/IzAVIrEBuvthQbp2kFZMCz9c95jr593I650s81GuaWf2MFiGy3utjQ"
      />
      <h3>
        London
      </h3>
      <div
        className="city__feature"
      >
        <Icon
          name="profile"
        />
        1200+ happy eaters
      </div>
      <div
        className="city__feature"
      >
        <Icon
          name="star"
        />
        50+ top chefs
      </div>
      <div
        className="city__feature"
      >
        <Icon
          name="twitter"
        />
        <a
          href=""
        >
          @omnifood_london
        </a>
      </div>
    </div>
  </div>
</section>
`;
